src/platforms/evdev/libinput_device.cpp

```diff
@@ -358,12 +361,26 @@ mir::EventUPtr mie::LibInputDevice::convert_touch_frame(libinput_event_touch* to
         if (data.action == mir_touch_action_down)
             data.action = mir_touch_action_change;
 
-        if (data.action == mir_touch_action_up)
+        if (data.action == mir_touch_action_up || data.down_notified)
             it = last_seen_properties.erase(it);
         else
             ++it;
+
+        data.down_notified = true;
```

Comment:
It is a bad idea to modify `data.down_notified` after (on one path) erasing the object referenced by `data`.